src/ServiceProvider.php

```diff
@@ -1,56 +1,71 @@
 <?php
 
-declare(strict_types=1);
+declare(strict_types = 1);
 
 namespace AvtoDev\RoadRunnerLaravel;
 
-use Illuminate\Contracts\Http\Kernel;
+use Illuminate\Foundation\Http\Kernel;
+use Illuminate\Contracts\Http\Kernel as KernelContract;
 
 class ServiceProvider extends \Illuminate\Support\ServiceProvider
 {
-    /**
-     * Indicates if loading of the provider is deferred.
-     *
-     * @var bool
-     */
-    protected $defer = false;
-
     /**
      * Register services and middleware.
      *
+     * @param KernelContract $kernel
+     *
      * @return void
      */
-    public function boot()
+    public function boot(KernelContract $kernel): void
     {
-        $this->registerForceHttpsMiddleware();
+        if ($kernel instanceof Kernel) {
+            // NOTE: Registering order is very important
+            // ForceHttpsMiddleware MUST be registered EARLIER then SetServerPortMiddleware
+            $this->registerSetServerPortMiddleware($kernel);
+            $this->registerForceHttpsMiddleware($kernel);
```

Comment:
Maybe it is worth mentioning that `Illuminate\Foundation\Http\Kernel::prependMiddleware()` is used to register middlwares? The discrepancy between the comment and the order of the called methods is confusing until you look inside the methods themselves.

Reply:
_updated_